src/App.js

```diff
@@ -1,15 +1,46 @@
+import { Route, Routes } from "react-router-dom";
+import { DataProvider } from "./contexts/DataContext";
 import "./App.css";
-import CoursesSection from "./components/coursesSection/CoursesSection.js";
-
-function App({ title, description, exploreButtonContent, courses }) {
-  return (
-    <CoursesSection
-      title={title}
-      description={description}
-      exploreButtonContent={exploreButtonContent}
-      courses={courses}
-    />
-  );
+import HomePage from "./components/homePage/HomePage";
+import CoursePage from "./components/coursePage/CoursePage";
+import NavBarSection from "./components/navbarSection/NavBarSection";
+import { useEffect, useState } from "react";
+
+function App() {
+  const [HomePageData, setHomePageData] = useState();
+  const [CourseData, setCourseData] = useState();
+
+  useEffect(() => {
+    fetch("https://api.npoint.io/73b880b5c0a10a42b9fb")
+      .then((response) => {
+        return response.json();
+      })
+      .then((json) => {
+        setHomePageData(json);
+      });
+
+    fetch("https://api.npoint.io/01485aaee696c8e359a6")
+      .then((response) => {
+        return response.json();
+      })
+      .then((json) => {
+        setCourseData(json);
+      });
+  }, []);
+
+  if (HomePageData === undefined || CourseData === undefined) {
```

Comment:
if any problem happened while fetching the data the loading indicator will be rendered forever.